server/src/main/java/org/opensearch/telemetry/tracing/TracerManager.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.telemetry.Telemetry;
+import org.opensearch.telemetry.TelemetrySettings;
+import org.opensearch.telemetry.tracing.noop.NoopTracer;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * TracerManager represents a single global class that is used to access tracers.
+ *
+ * The Tracer singleton object can be retrieved using TracerManager.getTracer(). The TracerManager object
+ * is created during class initialization and cannot subsequently be changed.
+ */
+public class TracerManager {
+
+    private static final Logger logger = LogManager.getLogger(TracerManager.class);
+    private static volatile TracerManager INSTANCE;
+
+    private volatile Tracer defaultTracer;
+    private final Object mutex = new Object();
+    private final TelemetrySettings telemetrySettings;
+    private final Telemetry telemetry;
+    private final ThreadPool threadPool;
+
+    /**
+     * Initializes the TracerFactory singleton instance
+     *
+     * @param telemetrySettings       tracer settings instance
+     * @param telemetry            telemetry instance
+     * @param threadPool           thread pool instance
+     */
+    public static synchronized void initTracerManager(TelemetrySettings telemetrySettings, Telemetry telemetry, ThreadPool threadPool) {
```

Comment:
Why it is implemented through statics? 
1) We have well defined place when TracerManager is created (Node)
2) The TracerManager should be used though standard injection mechanism (bind + @Inject) where needed

The `ThreadContext` uses `TracerManager` in a static way but we could get rid of that.

Reply:
This a trade off call we took to avoid code pollution. With bind+@inject approach we will have to inject tracer manager instance in all the classes where we need to instrument the code. Some of the classes are not even created through "guice" way and are instantiated with`new`. Hence the static handle would help them to get tracer instance without making any other changes.

Reply:
Removed static handle